db/compaction/compaction_picker_universal.cc

```diff
@@ -1150,8 +1205,13 @@ Compaction* UniversalCompactionBuilder::PickDeleteTriggeredCompaction() {
       if (sr->being_compacted) {
         continue;
       }
-      FileMetaData* f = vstorage_->LevelFiles(0)[loop];
+      FileMetaData* f = sr->file;
```

Comment:
[Note] A bug fix - cased by assuming sorted_run on L0 is always a prefix of vstorage->LevelFiles(0), which is no longer true after we sanitize sorted_run in this PR.